worked_examples/geospatial/workshop_april_2024/Workshop_AprilMay2024_part1.ipynb

```diff
@@ -0,0 +1,1360 @@
+   "id": "bbff216f",
+   "metadata": {},
+   "source": [
+    "> _**NOTE**_: We have used ``add_to_parents=False`` here so that these patches are not added to the ``my_files`` object. This is simply so that we don't have two sets of patches added after running both ``.patchify_all()`` commands."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4abbd4c0",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "my_files.patchify_all(method=\"pixel\", patch_size=1000, add_to_parents=False)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f60d6750",
+   "metadata": {},
+   "source": [
+    "#### Patchify by meters"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "1c1c383e",
+   "metadata": {},
+   "source": [
+    "You will only be able to use ``method=\"meters\"`` if you have coordinates saved for each of your map images. \n",
+    "These coordinates should correspond to the \"bounding box\" of your map image (minimum x, minimum y, maximum x and maximum y) and thereby associate the left, bottom, right and top edges of your map image to their geospatial locations.\n",
+    "\n",
+    "These can be added by running either ``.add_metadata()`` (assuming your metadata contains these coordinates) or, if your images contain georefencing information, ``.add_geo_info()``."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "99c8e6bf",
+   "metadata": {},
+   "source": [
+    "__**YOUR TURN**__: Patchify your maps into 100 meter patches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f33b361",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# my_files.patchify_all()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "9943557d",
+   "metadata": {},
+   "source": [
+    "You will see your patches are saved in separate directories, each indicating the patch size and method used."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "e05f0631",
+   "metadata": {},
+   "source": [
+    "### Visualize results"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "39297920",
+   "metadata": {},
+   "source": [
+    "``MapReader`` also contains some useful functions for visualizing your patches.\n",
+    "\n",
+    "For example, the ``.show_sample()`` method can be used to show a random sample of your patches:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "988bb6e2",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "my_files.show_sample(num_samples=3, tree_level=\"patch\", random_seed=1)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "d8a95b55",
+   "metadata": {},
+   "source": [
+    "You may also want to see all the patches created from one of your parent images. \n",
```

Comment:
This doesn't work for me. Parent images don't seem to have a parent_id assigned.

Error:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[117], line 1
----> 1 my_files.show_parent("map_75650661")

File ~/miniconda3/envs/mapreader_1.3/lib/python3.10/site-packages/mapreader/load/images.py:1506, in MapImages.show_parent(self, parent_id, column_to_plot, **kwargs)
   1476 def show_parent(
   1477     self,
   1478     parent_id: str,
   1479     column_to_plot: str | None = None,
   1480     **kwargs: dict,
   1481 ) -> None:
   1482     """
   1483     A wrapper method for `.show()` which plots all patches of a
   1484     specified parent (`parent_id`).
   (...)
   1504     :meth:`mapreader.load.images.MapImages.show` method for more detail.
   1505     """
-> 1506     patch_ids = self.parents[parent_id]["patches"]
   1507     figures = self.show(patch_ids, column_to_plot=column_to_plot, **kwargs)
   1509     return figures

KeyError: 'map_75650661'
```